lib/charms/layer/docker_registry.py

```diff
@@ -260,6 +265,28 @@ def _write_htpasswd(path, user, password):
     return True
 
 
+def _write_tls_blobs_to_files():
+    '''Write the user defined TLS blobs to files.
+
+    :return: None
+    '''
+    charm_config = hookenv.config()
+
+    blobs = [
+        ('tls-cert-blob', 'tls-cert-path'),
+        ('tls-ca-blob', 'tls-ca-path'),
+        ('tls-key-blob', 'tls-key-path')
+    ]
+
+    for blob, path in blobs:
+        blob = charm_config.get(blob)
+        path = Path(charm_config.get(path))
```

Comment:
We should check `config.get(path)` here to make sure it's truthy.  If the user provides a blob and clears the path, we'd be in for a wild ride:

```
juju config docker-registry tls-cert-blob=$(base64 /wow/eeeey) tls-cert-path=''
```

Reply:
I've pushed a change.  Is there any way to get the config default from the yaml?  Maybe it's better to fall back on that than just fail (or maybe it's better to fail to force the user into a conscious fix).